### DISCUSSION TITLE
Conflict with pydoc?

### DISCUSSION THREAD

**Discussion by UlrichGoebel** [Textualize/textual#4440] [Q&A]
Hi,

in a Textual application I try to use `docstrings` and `pydoc` to document my application. It seems that this doesn't work with Textual. What I tried:

`Test.py`
```python
#!/usr/bin/env python3

"""Test.py - Does nothing, just testing pydoc
"""

import textual

class MyClass():
    """MyClass - Does noting, just testing pydoc
    """
    def myMethod(self):
        """myMethod - Does nothing, just testing pydoc
        """
        pass

if __name__ == '__main__':
    myClass = MyClass()
```
`./Test.py` ends without error message.

But `pydoc ./Test.py` gives
```
problem in ./Test.py - ModuleNotFoundError: No module named 'textual'
```
Without the line `import textual` ist works. Even importing other modules `pydoc` works.

Did I overlook something in the docu? Any hints to a mistake I made?

**Comment by davep** [comment]
Assuming your Textual application works fine when you run it, but you only get this problem when running `pydoc`, I would imagine that you're using a virtual environment for running your application (and Textual is installed there) but that `pydoc` isn't being run from within that virtual environment, or it's doing something that causes work to take place that doesn't take the use of a virtual environment into account.

**Comment by UlrichGoebel** [comment]
Yes, I am using a virtual environment, and thanks to [davep](https://github.com/davep) hints a quick search confirms his thoughts:

https://stackoverflow.com/questions/58823295/pydoc-not-finding-installed-modules
https://github.com/pypa/virtualenv/issues/149
https://bugs.launchpad.net/virtualenv/+bug/315990

It seems to me that `pydoc` tries to import the code to be documented, and there is something to observe...

Unfortunately, I don't have the time right now to take a closer look. So, without further ado, I'm switching to the [Doxygen](https://www.doxygen.nl/index.html) tool. That works for me for the moment. May be I will come back to this later.

**Comment by TomJGooding** [comment]
Are you simply running `pydoc <name>` rather than `python -m pydoc <name>`?
